src/osmet/mod.rs

```diff
@@ -150,6 +151,23 @@ pub fn osmet_unpack(config: &OsmetUnpackConfig) -> Result<()> {
     Ok(())
 }
 
+pub fn find_matching_osmet_in_dir(sector_size: NonZeroU32, osmet_dir: &Path) -> Result<PathBuf> {
+    for entry in WalkDir::new(osmet_dir) {
+        let entry = entry?;
+
+        if !entry.file_type().is_file() {
+            continue;
+        }
+
+        let (osmet, _) = osmet_file_read(entry.path())?;
+        if osmet.sector_size == sector_size.get() {
+            return Ok(entry.into_path());
+        }
+    }
+
+    Err(format!("no matching osmet file found in {:?}", osmet_dir).into())
```

Comment:
Is this really a failure?  I'd think we'd return `None` and fall back to streaming in this case.

Reply:
Yeah, changed to return an `Option` instead. Relatedly, in #187, I've added a new `--offline` switch so that (1) we can test the offline path without first defaulting to it as done in this PR, and (2) one can assert that the install will be offline and error out otherwise. (Kinda like cargo's `--frozen` tag).